cpp/src/parquet/test_util.h

```diff
@@ -783,8 +789,33 @@ inline void GenerateData<ByteArray>(int num_values, ByteArray* out,
   random_byte_array(num_values, 0, heap->data(), out, 2, max_byte_array_len);
 }
 
+template <typename T>
+inline void GeneratePrefixedData(int num_values, T* out, std::vector<uint8_t>* heap,
+                                 double prefixed_probability);
+
+template <>
+inline void GeneratePrefixedData(int num_values, ByteArray* out,
+                                 std::vector<uint8_t>* heap,
+                                 double prefixed_probability) {
+  // seed the prng so failure is deterministic
+  int max_byte_array_len = 12;
+  heap->resize(num_values * max_byte_array_len);
```

Comment:
The comment would deserve being put just above the line it applies to :-)
```suggestion
  int max_byte_array_len = 12;
  heap->resize(num_values * max_byte_array_len);
  // seed the prng so failure is deterministic
```